### PR TITLE
Pin libraries and remove some dependancies

### PR DESCRIPTION
**Pull request by fayaaz**

### CODE REVIEW

File: requirements.txt [hunk 1]
```diff
@@ -2,16 +2,14 @@ azure-storage-blob==12.8.1
...
-colorama
-importlib-metadata
-keyring
-numpy
-pandas
-pathlib
-Pillow
+colorama==0.4.4
+importlib-metadata==4.8.1
+keyring==23.1.0
```
Comment (ahaith) [inline requirements.txt:7]: Not sure that these 3 are actually used. They don't show up in a search of the codebase.

**Review by ahaith — COMMENTED** [review]
Should we also pin all versions in setup.py, at least to major version?

**Comment by fayaaz** [comment]
It may be worth testing that it's ok to install this on Windows/Mac after the setup.py changes and everything working as expected

**Comment by tarqu1n** [comment]
@Vaspra are these changes ok?

**Review by ahaith — APPROVED** [review]
Let's just merge this. Can re-add anything that's needed.

**Pull request merged**